server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -407,7 +411,7 @@ private void handleIncomingRequest(final HttpRequest httpRequest, final HttpChan
             restRequest = innerRestRequest;
         }
 
-        final HttpTracer trace = tracer.maybeTraceRequest(restRequest, exception);
+        final HttpTracer maybeHttpTracer = httpTracer.maybeTraceRequest(restRequest, exception);
```

Comment:
I agree, it is confusing, and for a while I have `HttpTracer` implementing `Tracer`, but I ended up taking it as part of simplifying some stuff with Armin. The fundamental problem is that `HttpTracer` is actually just a logger, but we reference it in our docs so renaming the class isn't 100% trivial. However I've renamed many of the uses / instances of `HttpTracer` so make it clearer that it's about logging.

Reply:
Thanks for the rename of the member, it does help in the reading. However it is still confusing since the class is still called HttpTracer.

> we reference it in our docs so renaming the class isn't 100% trivial.

I think this could be handled through keeping the HttpTracer class, but with no impl, so that whatever the class is renamed to can still have `LogManager.getLogger(HttpTracer.class)`. We do need a way deprecate such logging; it is an existing problem that our class and package names leak into logger configuration since it restricts us from easily changing, or we accidentally make a configuration breaking change without realizing it.

The rename to HttpLogger I suggest above isn't crucial for this PR, but I do think it should be done as a followup.